dev-util/gdb/gdb-15.1.recipe

```diff
@@ -0,0 +1,116 @@
+SUMMARY="The GNU debugger"
+DESCRIPTION="GDB is the GNU debugger. It allows you to see what is going on \
+'inside' another program while it executes -- or what another program was \
+doing at the moment it crashed.
+
+GDB can do four main kinds of things (plus other things in support of \
+these) to help you catch bugs in the act:
+
+ * Start your program, specifying anything that might affect its behavior.
+ * Make your program stop on specified conditions.
+ * Examine what has happened, when your program has stopped.
+ * Change things in your program, so you can experiment with correcting \
+the effects of one bug and go on to learn about another.
+
+The program being debugged can be written in Ada, C, C++, Objective-C, \
+Pascal (and many other languages). Those programs might be executing \
+on the same machine as GDB (native) or on another machine (remote). \
+GDB can run on most popular UNIX and Microsoft Windows variants."
+HOMEPAGE="https://sourceware.org/gdb/"
+COPYRIGHT="2024 Free Software Foundation, Inc."
+LICENSE="GNU GPL v2
+	GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://ftp.gnu.org/gnu/gdb/gdb-$portVersion.tar.xz"
+CHECKSUM_SHA256="38254eacd4572134bca9c5a5aa4d4ca564cbbd30c369d881f733fb6b903354f2"
+PATCHES="gdb-$portVersion.patch"
+
+ARCHITECTURES="x86_64 ?x86 !x86_gcc2"
+
+binutilsVersion="2.42"
+
+PROVIDES="
+	gdb = $portVersion
+	cmd:gdb = $portVersion
+	cmd:gdb_add_index = $portVersion
+	cmd:gdbserver = $portVersion
+	devel:libbfd = $binutilsVersion
+	devel:libctf = $binutilsVersion
+	devel:libctf_nobfd = $binutilsVersion
+	devel:libopcodes = $binutilsVersion
+	devel:libsframe = $binutilsVersion
+	"
+
+REQUIRES="
+	haiku
+	lib:libexpat
+	lib:libgmp
+	lib:libiconv
+	lib:libisl
+	lib:libmpfr
+	lib:libncurses
+	lib:libpython3.10
+	lib:libreadline
+	lib:libz
+	"
+
+BUILD_REQUIRES="
+	devel:libexpat
+	devel:libgmp
+	devel:libiconv
+	devel:libisl
+	devel:libmpfr
+	devel:libncurses
+	devel:libpython3.10
+	devel:libreadline
+	devel:libz
+	"
+BUILD_PREREQUIRES="
+	haiku_devel
+	cmd:awk
+	cmd:bison
+	cmd:cmp
+	cmd:flex
+	cmd:gcc
+	cmd:ld
+	cmd:m4
+	cmd:make
+	cmd:makeinfo
+	cmd:python3
+	cmd:yacc
+	"
+
+TEST_REQUIRES="
+	cmd:runtest
+	"
+
+BUILD()
+{
+	runConfigure --omit-dirs "docDir dataRootDir" ./configure \
+		--disable-nls --enable-gdb --enable-gdbserver \
+		--with-system-zlib --with-system-readline
```

Comment:
I think you need a --with-python here (to ensure GDB actually picks up the python, if unspecified it may silently miss it.)

Reply:
It looks like we specifically need `--with-python=python3`.

Reply:
And apparently `export HAIKU_USE_VENDOR_DIRECTORIES=1` to tell Python to export the correct directories so GDB can find the headers.

Reply:
Resolved in new recipe.